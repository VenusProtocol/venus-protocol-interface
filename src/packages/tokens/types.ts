import { ChainId, Token } from 'types';

export type TokenMapping = {
  [chainId in ChainId]: Token[];
};
