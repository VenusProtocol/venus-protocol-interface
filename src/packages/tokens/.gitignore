# Automatically generated PancakeSwap token list
/infos/pancakeSwapTokens/bscMainnet.ts
