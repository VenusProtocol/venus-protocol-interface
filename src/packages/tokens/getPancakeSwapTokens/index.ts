import { ChainId } from 'types';

import pancakeSwapTokens from '../tokenInfos/pancakeSwap';

export interface GetPancakeSwapTokensInput {
  chainId: ChainId;
}

export const getPancakeSwapTokens = ({ chainId }: GetPancakeSwapTokensInput) =>
  pancakeSwapTokens[chainId];
