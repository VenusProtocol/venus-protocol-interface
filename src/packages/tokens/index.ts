export * from './tokenInfos';
export * from './utilities/getTokens';
export * from './utilities/getSwapTokens';
export * from './utilities/isTokenActionEnabled';
export * from './hooks/useGetToken';
export * from './hooks/useGetTokens';
export * from './hooks/useGetSwapTokens';
