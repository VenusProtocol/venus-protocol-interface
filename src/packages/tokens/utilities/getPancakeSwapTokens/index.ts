import { pancakeSwapTokens } from 'packages/tokens/tokenInfos/pancakeSwapTokens';
import { ChainId } from 'types';

export interface GetPancakeSwapTokensInput {
  chainId: ChainId;
}

export const getPancakeSwapTokens = ({ chainId }: GetPancakeSwapTokensInput) =>
  pancakeSwapTokens[chainId];
