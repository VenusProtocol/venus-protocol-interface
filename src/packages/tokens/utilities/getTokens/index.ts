import { tokens } from 'packages/tokens/tokenInfos/commonTokens';
import { ChainId } from 'types';

export interface GetTokensInput {
  chainId: ChainId;
}

export const getTokens = ({ chainId }: GetTokensInput) => tokens[chainId];
