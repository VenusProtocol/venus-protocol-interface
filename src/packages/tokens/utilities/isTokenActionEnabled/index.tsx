import disabledTokenActions from 'packages/tokens/tokenInfos/disabledTokenActions';
import { ChainId, TokenAction } from 'types';

import areAddressesEqual from 'utilities/areAddressesEqual';

export interface IsTokenActionEnabledInput {
  tokenAddress: string;
  chainId: ChainId;
  action: TokenAction;
}

export const isTokenActionEnabled = ({
  tokenAddress,
  chainId,
  action,
}: IsTokenActionEnabledInput) => {
  const disabledToken = disabledTokenActions[chainId].find(item =>
    areAddressesEqual(item.address, tokenAddress),
  );

  return !disabledToken || !disabledToken.disabledActions.includes(action);
};
