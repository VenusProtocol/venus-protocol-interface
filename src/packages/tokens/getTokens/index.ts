import { ChainId } from 'types';

import tokens from '../tokenInfos/common';

export interface GetTokensInput {
  chainId: ChainId;
}

export const getTokens = ({ chainId }: GetTokensInput) => tokens[chainId];
