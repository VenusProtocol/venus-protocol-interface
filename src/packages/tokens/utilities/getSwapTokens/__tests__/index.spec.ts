import Vi from 'vitest';

import { busd, vai, vrt, xvs } from '__mocks__/models/tokens';

import { getPancakeSwapTokens } from 'packages/tokens/utilities/getPancakeSwapTokens';
import { getTokens } from 'packages/tokens/utilities/getTokens';
import { ChainId } from 'types';

import { getSwapTokens } from '..';

vi.mock('packages/tokens/utilities/getTokens');
vi.mock('packages/tokens/utilities/getPancakeSwapTokens');

const fakeTokens = [xvs, vai];
const fakePancakeSwapTokens = [vrt, xvs, busd];

describe('getSwapTokens', () => {
  beforeEach(() => {
    (getTokens as Vi.Mock).mockImplementation(() => fakeTokens);
    (getPancakeSwapTokens as Vi.Mock).mockImplementation(() => fakePancakeSwapTokens);
  });

  it('returns a list of tokens that contains all the tokens listed on Pancake Swap and all the tokens listed on Venus with no duplicates', () => {
    const result = getSwapTokens({
      chainId: ChainId.BSC_TESTNET,
    });

    expect(result).toMatchSnapshot();
  });
});
