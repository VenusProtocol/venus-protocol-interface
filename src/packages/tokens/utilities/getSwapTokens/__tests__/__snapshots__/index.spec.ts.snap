// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getSwapTokens > returns a list of tokens that contains all the tokens listed on Pancake Swap and all the tokens listed on Venus with no duplicates 1`] = `
[
  {
    "address": "0x331F639B4F3CF6E391CD244e0b5027C5968Ec325",
    "asset": "/src/packages/tokens/img/vrt.svg",
    "decimals": 18,
    "symbol": "VRT",
  },
  {
    "address": "0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47",
    "asset": "/src/packages/tokens/img/busd.svg",
    "decimals": 18,
    "symbol": "BUSD",
  },
  {
    "address": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
    "asset": "/src/packages/tokens/img/xvs.svg",
    "decimals": 18,
    "symbol": "XVS",
  },
  {
    "address": "0x5fFbE5302BadED40941A403228E6AD03f93752d9",
    "asset": "/src/packages/tokens/img/vai.svg",
    "decimals": 18,
    "symbol": "VAI",
  },
]
`;
